### DISCUSSION TITLE
SSI diverges from Policy Rules Database dashboard

### DISCUSSION THREAD

**Discussion by MaxGhenis** [PolicyEngine/policyengine-us#1472] [General]
# Summary

PolicyEngine produces different SSI estimates than the [Atlanta Fed's Policy Rules Database dashboard](https://emar-data-tools.shinyapps.io/prd_dashboard/) for two cases I examined. In the first, I believe we're correct, as the calculation is relatively straightforward. I didn't run the full numbers for the second case, which involves parent-to-child deeming, though [this notebook](https://colab.research.google.com/drive/1KJzkhuFO2h4YEbBKEDn_Pw22Q22_AzHe#scrollTo=tfq8_AppDyrk) shows both computation trees.

[Here's the PRD SSI code.](https://github.com/FRB-Atlanta-Advancing-Careers/policy-rules-database/blob/main/functions/benefits_functions.R#L1252)

_Annual numbers_
| Scenario | PolicyEngine | [PRD dashboard](https://emar-data-tools.shinyapps.io/prd_dashboard/) |
| - | - | - |
| Single disabled person with $18k earnings | $1,602 | $3,993 |
| Single parent of a disabled child with $30k earnings | $5,934 | $882 |

## Single disabled person with $18k earnings:
<img width="545" alt="image" src="https://user-images.githubusercontent.com/6076111/198878669-b32ea066-ee28-42d1-99d5-a1ed8e162f02.png">
<img width="1437" alt="image" src="https://user-images.githubusercontent.com/6076111/198878655-224e9a82-8b37-4c96-9c32-442e2d4f9289.png">

$18k/year = $1,500/month
Countable income = ($1,500 - $20 general exclusion - $65 earnings exclusion) * 50% after % earnings exclusion = $707.50
Monthly benefit = $841 - $707.50 = $133.50
Annual benefit = $133.50 * 12 = $1,602

## Single non-disabled parent of a disabled child with $30k income:
<img width="565" alt="image" src="https://user-images.githubusercontent.com/6076111/198878406-e099448f-52ba-462e-a35c-c4c6b18e53b4.png">
<img width="1459" alt="image" src="https://user-images.githubusercontent.com/6076111/198878502-a2bfba96-ac97-4b54-9e00-cb3346641fe9.png">

cc @eliasilin

**Comment by ghost** [comment]
@MaxGhenis Here are our responses

# Single disabled person with $18k earnings:
Seems like in this calculation, you are not counting a pretty generous California State Supplemental Payment (SSP) of $199.21/m. PRD adds SSP to the overall SSI amount, rather than showing it as a separate program.

$18k/year = $1,500/month
Countable income = ($1,500 - $20 general exclusion - $65 earnings exclusion) * 50% after % earnings exclusion = $707.50
Monthly benefit = $841 + **199.21**  - $707.50 = **$332.71**
Annual benefit = **$332.71*** 12 = **$3,992.52**

# Single non-disabled parent of a disabled child with $30k income:

We reconfirmed the value produced by the PRD Dashboard. More digging is required to figure out the source of discrepancy.

Total countable income = 30k or $2,500/month
Deemed earned income = [2,500 - 20 income disregard - 65 earnings disregard] * .50 = $1,207.50 - **$841 parental allocation**
= **$366.50** Deemed income for the child which is considered unearned income for the child. The income disregard is applied to unearned income and this new total is subtracted from the difference in FBR amount between a married couple and an individual.

$420 FBR difference - [**366.50** - 20 income disregard] = 73.5 * 12 = **$882**

For clarification the $841 parental allocation just means they subtract the monthly FBR amount for a couple or individual. "Parental Allocation" is what we called it in the PRD technical manual.

Note, that even though the parent is non-disabled, $841 still should be subtracted. See the SSA rules here. Part (d)(i)(ii). Ineligible parents here should refer to parents who do not receive SSI 
https://www.ssa.gov/OP_Home/cfr20/416/416-1165.htm

**Reply by MaxGhenis** [reply]
Thank you! Currently we only have the MA SSP. I changed the PRD dashboard to WA, which doesn't really have an SSP, and got the same $1,602 as we have. ✅ 

<img width="1438" alt="image" src="https://user-images.githubusercontent.com/6076111/199049497-219813a5-22ad-49d8-a127-cdd856288826.png">

In WA, the second case still produces $882. We'll dig into our logic again, helpful to see yours.

<img width="1441" alt="image" src="https://user-images.githubusercontent.com/6076111/199049749-b1712963-4a10-4bc2-b2a4-fda0ca657998.png">
